compiler/noirc_evaluator/src/ssa/opt/remove_bit_shifts.rs

```diff
@@ -79,6 +79,8 @@ impl Function {
 
             context.replace_value(old_result, new_result);
         });
+
+        debug_assert!(remove_bit_shifts_post_check(self));
```

Comment:
Perhaps instead of returning a bool, we could create a macro that calls the function only if `#[cfg(debug_assertions)]` as suggested [here](https://users.rust-lang.org/t/show-value-only-in-debug-mode/43686/3), and then we can include a bit more information in the panic.

Reply:
I did that in https://github.com/noir-lang/noir/pull/8481